bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -236,21 +239,21 @@ static boolean isConvertibleToFloatingPointTypes(Object value) {
     }
 
     // TODO: return only the first matching type
-    public static List<Type> getConvertibleTypes(Object inputValue, Type targetType) {
+    public static Set<Type> getConvertibleTypes(Object inputValue, Type targetType) {
         return getConvertibleTypes(inputValue, targetType, new ArrayList<>());
```

Comment:
The following still doesn't work.

```ballerina
    float[] x = [1, 2];
    (int|byte)[] y = checkpanic x.cloneWithType();
```

Reply:
Need to check with combinations like int and int subtypes too.

Reply:
Addressed with https://github.com/ballerina-platform/ballerina-lang/pull/30710/commits/48f5e12c00c98975124f5e2a952137570e4d8579